### ISSUE TITLE
rule.UnexportedReturn: false positive when return type is exported but an alias to an unexported type

### ISSUE THREAD

**Issue by dolmen** [mgechev/revive#1406]
**Describe the bug**

Here is an example code that implements functional options, but uses type aliases to hide implementation details.

```go
// Command main...
package main

type config struct {
	N int
	// ...
}

// Option is an option for [New].
type Option = option

// option is the hidden concrete implementation of Option.
type option func(*config)

// WithN sets N.
func WithN(n int) Option { // <------ This should not trigger Unexported Return
	return func(c *config) {
		c.N = n
	}
}

// T is solid.
type T struct{}

// New builds a [T] from [Option]s.
func New(opts ...Option) T {
	var c config
	for _, opt := range opts {
		opt(&c)
	}
	return T{}
}
```

**To Reproduce**

```console
$ golangci-lint run -E revive
option.go:16:19: unexported-return: exported func WithN returns unexported type main.Option, which can be annoying to use (revive)
func WithN(n int) Option { // <------ This should not trigger Unexported Return
                  ^
1 issues:
* revive: 1
* $ golangci-lint version      
golangci-lint has version 2.1.6 built with go1.24.2 from eabc2638 on 2025-05-04T15:41:19Z
$ go version -m $(which golangci-lint) | grep revive
	dep	github.com/mgechev/revive	v1.9.0	h1:8LaA62XIKrb8lM6VsBSQ92slt/o92z5+hTw3CmrvSrM=
```

-->

**Expected behavior**

A function that returns an exported type (`Option` in the example) should not trigger the UnexportedReturn rule. Even if the type is an alias to an unexported type (`option` in the example).

**Comment by dolmen** [comment]
Here is code in the wild impacted by this issue:
https://github.com/DataDog/dd-trace-go/blob/604795a9ed5039f803477cb822d945eb0dede6c1/contrib/gin-gonic/gin/option.go#L51
